### PR TITLE
Make framework buildspecs pointers to most recent buildspecs

### PR DESCRIPTION
**Pull request by arjkesh**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Tests run
- pt training, tf training, tf inference + sanity tests ([9fedcf4](https://github.com/aws/deep-learning-containers/pull/3946/commits/9fedcf403b47347d27d8a4591dc692802638f328))
- pt inference ([7d93894](https://github.com/aws/deep-learning-containers/pull/3946/commits/7d93894996f6c934f1693b9c33df1ffa8c294a41)) --> safety test fails, ignoring
- tf/pt inference graviton ([96d68d4](https://github.com/aws/deep-learning-containers/pull/3946/commits/96d68d4031f5c40a5c7db60aa00b46d870621b0f))

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [ ] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**
<details>
<summary>Expand</summary>

- [ ] `sagemaker_remote_tests = true`
- [ ] `sagemaker_efa_tests = true`
- [ ] `sagemaker_rc_tests = true`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

</details>

### Formatting
- [ ] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

#### Builds to Execute
<details>
<summary>Expand</summary>

Click the checkbox to enable a build to execute upon merge.

*Note: By default, pipelines are set to "latest". Replace with major.minor framework version if you do not want "latest".*

- [ ] build_pytorch_training_latest
- [ ] build_pytorch_inference_latest
- [ ] build_tensorflow_training_latest
- [ ] build_tensorflow_inference_latest

</details>

### PR Checklist 
<details>
<summary>Expand</summary>

- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

#### NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ec2_benchmark_tests = true` or `sagemaker_benchmark_tests = true`
</details>

### Pytest Marker Checklist
<details>
<summary>Expand</summary>

- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type
</details>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/image_builder.py [hunk 1]
```diff
@@ -80,6 +81,14 @@ def image_builder(buildspec, image_types=[], device_types=[]):
     :param image_types: <list> list of image types
     :param device_types: <list> list of image device type
     """
+    with open(buildspec, "r") as bf:
+        pointer_check = yaml.safe_load(bf)
+        pointer = pointer_check.get("buildspec_pointer")
+        if pointer:
+            FORMATTER.print(f"Buildspec {buildspec} points to another buildspec file {pointer}")
```
Comment (arjkesh) [inline src/image_builder.py:88]: check for build context if this is a pointer. In PRs this is ok, elsewhere we will want a failure

File: src/image_builder.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 import re
 import json
 import tempfile
+import yaml
```
Comment (arjkesh) [inline src/image_builder.py:22]: remove

File: src/image_builder.py [hunk 3]
```diff
@@ -627,6 +625,13 @@ def tag_image_with_datetime(image_tag):
     return f"{image_tag}-{datetime_suffix}"
 
 
+def tag_image_with_initiator(image_tag):
+    """
+    Add additional debug tags
+    """
+    return f"{image_tag}-{os.getenv('CODEBUILD_INITIATOR', '').split('/')[-1]}-{os.getenv('CODEBUILD_RESOLVED_SOURCE_VERSION', '')[:7]}"
```
Comment (arjkesh) [inline src/image_builder.py:632]: rework this to put initiator at the end

File: pytorch/inference/buildspec-graviton-2-2.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      target: config.properties
+
+images:
+  BuildEC2PytorchGravitonCPUInferencePy3DockerImage:
+    <<: *INFERENCE_REPOSITORY
+    build: &PYTORCH_CPU_INFERENCE_PY3 false
+    image_size_baseline: 10000
+    device_type: &DEVICE_TYPE cpu
+    os_version: &OS_VERSION ubuntu20.04
+    torch_serve_version: &TORCHSERVE_VERSION 0.10.0
```
Comment (arjkesh) [inline pytorch/inference/buildspec-graviton-2-2.yml:41]: get latest

File: pytorch/inference/buildspec-2-1-sm.yml [hunk 5]
```diff
@@ -5,7 +5,7 @@ framework: &FRAMEWORK pytorch
 version: &VERSION 2.1.0
 short_version: &SHORT_VERSION "2.1"
 arch_type: x86
-autopatch_build: "False"
+autopatch_build: "True"
```
Comment (arjkesh) [inline pytorch/inference/buildspec-2-1-sm.yml:8]: @junpuf - enabling autopatch again on these images

File: pytorch/inference/buildspec-2-1-ec2.yml [hunk 6]
```diff
@@ -5,7 +5,7 @@ framework: &FRAMEWORK pytorch
 version: &VERSION 2.1.0
 short_version: &SHORT_VERSION "2.1"
 arch_type: x86
-autopatch_build: "False"
+autopatch_build: "True"
```
Comment (arjkesh) [inline pytorch/inference/buildspec-2-1-ec2.yml:8]: @junpuf - enabling autopatch again on these images

**Review by sallyseok — APPROVED** [review]

**Pull request merged**